File: javascript/high-scores/high-scores.js
// @ts-nocheck

/**
 * @constructor
 * @param {number[]} scores
 */
export function HighScores(scores) {
  this.latest = (this.scores = scores.slice()).at(-1);
  this.personalBest = Math.max(...scores);
  this.personalTopThree = scores.toSorted((a, b) => b - a).slice(0, 3); }

File: fortran/high-scores/high_scores.f90
MODULE high_scores; IMPLICIT NONE; CONTAINS
  PURE FUNCTION personalTopThree(arr) RESULT(top); INTEGER, INTENT(IN) :: arr(:)
    INTEGER top(3), i, p; LOGICAL(1) m(size(arr)); m = .TRUE.; DO i = 1, 3
      p = maxloc(arr, 1, m); top(i) = arr(p); m(p) = .FALSE.; END DO; END

  PURE FUNCTION scores(arr); INTEGER, INTENT(IN) :: arr(:)
    INTEGER scores(size(arr)); scores = arr; END FUNCTION

  PURE INTEGER FUNCTION latest(arr); INTEGER, INTENT(IN) :: arr(:)
    latest = arr(size(arr)); END FUNCTION

  PURE INTEGER FUNCTION personalBest(arr); INTEGER, INTENT(IN) :: arr(:)
    personalBest = maxval(arr); END FUNCTION; END MODULE
